### DISCUSSION TITLE
How to compile esp32_basic_synth

### DISCUSSION THREAD

**Discussion by m1cha1s** [marcel-licence/esp32_basic_synth#28] [Q&A]
Hello, I'm currently having a very hard time figuring out how do I compile the project. The project consists of multiple .ino files and when I open the main file in Arduino ide it only compiles and links it and not the rest does anyone know the answer?

**Comment by marcel-licence** [comment]
Hello,
your approach is fully correct. Arduino does create automatically a cpp out of all .ino files which you wouldn't see. This is the file which will be compiled and linked. Finally when there is not error you successfully compiled the whole project.
Hope this helps.
Best regards.

**Reply by m1cha1s** [reply]
Hmm, then in my case when I compile esp32_basic_synth.ino file with setting to use internal DAC it throws out this: 'writeDAC' was not declared in this scope. By the way, I changed 'l_sample' to 'fl_sample'.

**Reply by marcel-licence** [reply]
The NODAC or the option to use the internal mechnism is actually a left-over. I didn't maintain it yet because there are lots of possibilities to modify this project but at the actual state I could take a look again because the i2s part is nearly at is final stage =)

**Reply by m1cha1s** [reply]
When I change back to use external DAC it shows this error: analogSetCycles' was not declared in this scope

**Reply by marcel-licence** [reply]
As mentioned in other discussions I used an older version of the esp32 lib. I've also updated the readme.
I've commited a refreshed version to allow using "NO DAC"

**Reply by ErichHeinemann** [reply]
Hi I have seen this error: analogSetCycles too. You have to install the fresh ESP32-Arduino-Port.
I deleted the old folder and did a fresh install. Afterwards I was able to play around with the DAC-Options  for the ESp32.
